commit 15cb0242ea58ae0c102fc84c3044b4c2f5fc532c
Author: Daniel Hofstetter <daniel.hofstetter@42dh.com>
Date:   Mon Mar 11 16:24:39 2024 +0100

    uucore/pipes: adapt to new return type of nix fn
    
    nix 0.28 changed the return type of unistd::pipe() from Result<(RawFd, RawFd), Error> to Result<(OwnedFd, OwnedFd), Error>

Index: coreutils/src/uucore/src/lib/features/pipes.rs
===================================================================
--- coreutils.orig/src/uucore/src/lib/features/pipes.rs
+++ coreutils/src/uucore/src/lib/features/pipes.rs
@@ -8,6 +8,7 @@ use std::fs::File;
 use std::io::IoSlice;
 #[cfg(any(target_os = "linux", target_os = "android"))]
 use std::os::unix::io::AsRawFd;
+use std::os::unix::io::FromRawFd;
 
 #[cfg(any(target_os = "linux", target_os = "android"))]
 use nix::fcntl::SpliceFFlags;
@@ -20,7 +21,8 @@ pub use nix::{Error, Result};
 /// from the first.
 pub fn pipe() -> Result<(File, File)> {
     let (read, write) = nix::unistd::pipe()?;
-    Ok((File::from(read), File::from(write)))
+    // SAFETY: The file descriptors do not have other owners.
+    unsafe { Ok((File::from_raw_fd(read), File::from_raw_fd(write))) }
 }
 
 /// Less noisy wrapper around [`nix::fcntl::splice`].
